test/integration/authorise_application_test.rb

```diff
@@ -6,12 +6,17 @@ class AuthoriseApplicationTest < ActionDispatch::IntegrationTest
     @user = create(:user)
   end
 
-  should "not confirm the authorisation for a signed-out user" do
-    visit "/oauth/authorize?response_type=code&client_id=#{@app.uid}&redirect_uri=#{@app.redirect_uri}"
-    assert_response_contains("You need to sign in")
-    signin(@user)
+  should "not confirm the authorisation until the user signs in" do
+    begin
+      visit "/oauth/authorize?response_type=code&client_id=#{@app.uid}&redirect_uri=#{@app.redirect_uri}"
+      assert_response_contains("You need to sign in")
+      signin(@user)
+    rescue ActionController::RoutingError
```

Comment:
During testing, requests for all domains get routed to Signon; including the capybara browser being redirected to other apps.

The browser gets a redirect to url of the destination app. This then gets routed to Signon but Signon doesn't know how to handle the route.

Should I add a small comment in the test similar to the above explanation?